sympy/matrices/expressions/matpow.py

```diff
@@ -71,7 +71,7 @@ def doit(self, **kwargs):
             raise ValueError("Matrix determinant is 0, not invertible.")
         elif isinstance(base, (Identity, ZeroMatrix)):
             return base
-        elif isinstance(base, MatrixBase) and exp.is_number:
+        elif isinstance(base, MatrixBase) and (exp.is_number or exp.is_negative is not None or base.det() != 0):
```

Comment:
Is this change a crossover from the other PR?

Reply:
#17175, perhaps